### ISSUE TITLE
Unable to start localstack [ ERROR:localstack.services.generic_proxy ] 

### ISSUE THREAD

**Issue by akmalrasool** [localstack/localstack#918]
Hi
Hope you are doing well.
I have recently installed localstack but right after installation I hit **localstack start** and got following error:

```
$ localstack start
Starting local dev environment. CTRL-C to quit.
Starting mock S3 (http port 4572)...
Starting mock SSM (http port 4583)...
Starting mock API Gateway (http port 4567)...
Starting mock DynamoDB (http port 4569)...
Starting mock DynamoDB Streams service (http port 4570)...
Starting mock Lambda service (http port 4574)...
Starting mock CloudFormation (http port 4581)...
Starting mock CloudWatch (http port 4582)...
2018-09-04T19:29:05:ERROR:localstack.services.generic_proxy: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4563): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001871CED1F98>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it',)) Traceback (most recent call last):
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\connection.py", line 141, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\util\connection.py", line 83, in create_connection
    raise err
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\util\connection.py", line 73, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [WinError 10061] No connection could be made because the target machine actively refused it

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\connectionpool.py", line 357, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\http\client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\http\client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\http\client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\http\client.py", line 1026, in _send_output
    self.send(msg)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\http\client.py", line 964, in send
    self.connect()
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\connection.py", line 166, in connect
    conn = self._new_conn()
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\connection.py", line 150, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x000001871CED1F98>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\requests\adapters.py", line 440, in send
    timeout=timeout
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\connectionpool.py", line 639, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\urllib3\util\retry.py", line 388, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=4563): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001871CED1F98>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\localstack\services\generic_proxy.py", line 201, in forward
    headers=forward_headers)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\requests\api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\requests\api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\requests\sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\requests\sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\requests\adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=4563): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x000001871CED1F98>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it',))

2018-09-04T19:30:05:ERROR:localstack.services.generic_proxy: Exception running proxy on port 4581: [WinError 10038] An operation was attempted on something that is not a socket Traceback (most recent call last):
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\localstack\services\generic_proxy.py", line 291, in run_cmd
    self.httpd.serve_forever()
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\socketserver.py", line 236, in serve_forever
    ready = selector.select(poll_interval)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\selectors.py", line 323, in select
    r, w, _ = self._select(self._readers, self._writers, [], timeout)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\selectors.py", line 314, in _select
    r, w, x = select.select(r, w, w, timeout)
OSError: [WinError 10038] An operation was attempted on something that is not a socket

2018-09-04T19:30:05:ERROR:localstack.services.generic_proxy: Exception running proxy on port 4569: [WinError 10038] An operation was attempted on something that is not a socket Traceback (most recent call last):
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\site-packages\localstack\services\generic_proxy.py", line 291, in run_cmd
    self.httpd.serve_forever()
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\socketserver.py", line 236, in serve_forever
    ready = selector.select(poll_interval)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\selectors.py", line 323, in select
    r, w, _ = self._select(self._readers, self._writers, [], timeout)
  File "C:\users\akmal\appdata\local\programs\python\python36\lib\selectors.py", line 314, in _select
    r, w, x = select.select(r, w, w, timeout)
OSError: [WinError 10038] An operation was attempted on something that is not a socket
```

Any help will be much appericiated.
Thanks
AkmalRasool

**Comment by ss207210** [comment]
Getting similar error

**Comment by mwarkentin** [comment]
Seeing the same on latest w/ persistence and kinesis/firehose services in docker, rolled back to 0.8.0 and things started working again.

Previous error:

```
localstack_1    | 2019-01-16T01:46:03:ERROR:localstack.services.generic_proxy: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4565): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ffbb3062710>: Failed to establish a new connection: [Errno 111] Connection refused',)) Traceback (most recent call last):
localstack_1    |   File "/opt/code/localstack/localstack/services/generic_proxy.py", line 216, in forward
localstack_1    |     headers=forward_headers, stream=True)
localstack_1    |   File "/opt/code/localstack/.venv/lib/python2.7/site-packages/requests/api.py", line 116, in post
localstack_1    |     return request('post', url, data=data, json=json, **kwargs)
localstack_1    |   File "/opt/code/localstack/.venv/lib/python2.7/site-packages/requests/api.py", line 60, in request
localstack_1    |     return session.request(method=method, url=url, **kwargs)
localstack_1    |   File "/opt/code/localstack/.venv/lib/python2.7/site-packages/requests/sessions.py", line 533, in request
localstack_1    |     resp = self.send(prep, **send_kwargs)
localstack_1    |   File "/opt/code/localstack/.venv/lib/python2.7/site-packages/requests/sessions.py", line 646, in send
localstack_1    |     r = adapter.send(request, **kwargs)
localstack_1    |   File "/opt/code/localstack/.venv/lib/python2.7/site-packages/requests/adapters.py", line 516, in send
localstack_1    |     raise ConnectionError(e, request=request)
localstack_1    | ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=4565): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ffbb3062710>: Failed to establish a new connection: [Errno 111] Connection refused',))
localstack_1    |
localstack_1    | 2019-01-16T01:46:03:WARNING:infra.pyc: Service "kinesis" not yet available, retrying...
```

**Comment by FelikZ** [comment]
Issue was introduced in 0.8.7, **downgrade to 0.8.6 resolves** issue for me.

I did some investigation and in my case it is seems to be caused by crashing / stopped `kinesalite`:
https://github.com/localstack/localstack/blob/fa1a64c954b89b88ac30e77fd12930efc04c04c5/localstack/services/kinesis/kinesis_starter.py#L22

If I docker exec to container and start `kinesalite` by hands - things starting to work even in latest localstack.

See in changelog:

> 0.8.7 bump kinesalite version

This could be a root cause.

Related issues:
https://github.com/localstack/localstack/issues/230
https://github.com/localstack/localstack/issues/894

**Comment by whummer** [comment]
Thanks for reporting @akmalrasool @mwarkentin @FelikZ . This has potentially been fixed in #1526 - any chance you can create a test case to reproduce this issue, and see whether it is fixed with the latest version? Thanks!

**Comment by rockywarren-vertex** [comment]
I still need the workaround even after #1526.

**Comment by whummer** [comment]
@rockywarren-vertex Can you please start LocalStack with debugging (`DEBUG=1`) enabled, and post the console output here? Thanks

**Comment by ballwood** [comment]
Issue seems to be with DATA_DIR in docker, as soon as I disable it everything works fine. 

Maybe indicates running/mounting permissions mismatch?

**Comment by whummer** [comment]
Should be fixed in #1711 - please report here if the problem persists. Thanks
